src/ResultDecoder.php

```diff
@@ -282,4 +300,36 @@ public function transformQuotesSumamary(string $responseBody): array
 
         return $decoded['quoteSummary']['result'];
     }
+
+    public function transformOptions(string $responseBody): array
+    {
+        $decoded = json_decode($responseBody, true);
+        if (!isset($decoded['optionChain']['result']) || !\is_array($decoded['optionChain']['result'])) {
+            throw new ApiException('Yahoo Search API returned an invalid result.', ApiException::INVALID_RESPONSE);
+        }
+
+        $results = $decoded['optionChain']['result'];
+
+        // Single element is returned directly in "quote"
+        return array_map(function (array $item) {
+            return $this->createOption($item);
+        }, $results);
+    }
```

Comment:
I'd like to have a test case similar to `transformQuotes_jsonGiven_createArrayOfQuote` for this method. So we can ensure all expected fields are properly mapped.

Reply:
PR is updated with additional tests for the options result